documentation/restarting.markdown

```diff
@@ -17,13 +17,29 @@ There are several ways to send a HUP signal to the Puppet Server process, but th
 
     kill -HUP `pgrep -f puppet-server`
 
+Starting in version 2.7.0, you can also reload Puppet Server by running the "reload" action via the operating system's service framework.  This is analogous to sending a hangup signal but with the benefit of having the "reload" command pause until the server has been completely reloaded, similar to how the "restart" command pauses until the service process has been fully restarted.
```

Comment:
> Do you think any of the above would be useful to call out here? I didn't want to be unnecessarily wordy here if some of these benefits would be obvious to most users.

For the users where it isn't immediately obvious, I think we can at least note that `reload` doesn't require users to track the `puppetserver` process's PID, has more consistent and predictable results, and returns a more informative exit code.

> I'd be good with saying something about how we recommend the use of service reload in automated tooling because it more reliably determines when/whether the reload is successful

👍  to that.